src/NetscriptPort.ts

```diff
@@ -42,9 +42,10 @@ export function writePort(n: PortNumber, value: unknown): PortData | null {
       `port.write: Tried to write type ${typeof value}. Only string and number types may be written to ports.`,
     );
   }
-  const { data, resolvers } = getPort(n);
+  let { data, resolvers } = getPort(n);
   data.push(value);
-  while (resolvers.length > 0) resolvers.pop()?.();
+  for (const res of resolvers) res();
+  resolvers = [];
```

Comment:
As written this does not clear out old resolvers from the port. The new empty array is assigned to the `resolvers` variable from this function, but this does not eliminate any resolvers stored on the port.

Reply:
once the resolvers are called and then the array they are stored in is left for garbage collection is that not enough?

Reply:
Should be fixed now.